src/experimental/tools/download2go/index.ts

```diff
@@ -0,0 +1,415 @@
+            db.put("manifests", {
+              contentID,
+              transport,
+              manifest,
+              builder: { video, audio, text },
+              progress,
+              size,
+              duration,
+              metaData,
+            }).then(() =>
+              this.trigger("insertDB", {
+                action: "store",
+                contentID,
+                progress: progress.percentage,
+              })
+            ).catch((error: Error) => {
+              this.trigger("error", {
+                action: "resume-downloader",
+                error,
+                contentID,
+              });
+            });
+          },
+          (error: Error) =>
+            this.trigger("error", {
+              action: "resume-downloader",
+              error,
+              contentID,
+            })
+        );
+      this.activeDownloads[contentID] = resumeDownloadSub;
+    } catch (err) {
+      const error = err as Error;
+      this.trigger("error", { action: "resume-downloader", error, contentID });
+    }
+  }
+
+  /**
+   * Pause a download already started earlier.
+   * @param {string} contentID
+   * @returns {void}
+   */
+  pause(contentID: string): void {
+    try {
+      checkForPauseAMovie(contentID);
+      const activeDownloads = this.activeDownloads;
+      const activePauses = this.activePauses;
+      if (activeDownloads[contentID] == null && activePauses[contentID] == null) {
+        throw new ValidationArgsError(`Invalid contentID given: ${contentID}`);
+      }
+      activePauses[contentID].next();
+      activePauses[contentID].complete();
+      activeDownloads[contentID].unsubscribe();
+      delete activeDownloads[contentID];
+      delete activePauses[contentID];
+    } catch (e) {
+      const error = e as Error | undefined;
+      this.trigger("error", {
+        action: "pause",
+        contentID,
+        error: error !== undefined ? error : new Error("A Unexpected error happened"),
+      });
+    }
+  }
+
+  /**
+   * Get all the downloaded entry (manifest) partially or fully downloaded.
+   * @returns {Promise.<IStoredManifest[]|void>}
+   */
+  getAllOfflineContent(
+    limit?: number
+  ): Promise<IStoredManifest[] | undefined> | void {
+    try {
+      if (this.db == null) {
+        throw new Error("The IndexDB database has not been created!");
+      }
+      // TODO: return formatted content ?
+      return this.db.getAll("manifests", undefined, limit);
+    } catch (e) {
+      const error = e as Error;
+      this.trigger("error", {
+        action: "getAllDownloadedMovies",
+        error: new IndexDBError(error.message),
+      });
+    }
+  }
+
+  /**
+   * Get a singleMovie ready to be played by the rx-player,
+   * could be fully or partially downloaded.
+   * @param {string} contentID
+   * @returns {Promise.<IContentLoader|void>}
+   */
+  async getSingleContent(contentID: string): Promise<IContentLoader | void> {
```

Comment:
IContentLoader? Why all that?
Why does it need to know about the `contentProtection` object?

Reply:
He does need to know about the `contentProtection` to be able to `load()` the **sessionsIDS** again during the playback as follow:
```js
licenseStorage: {
  save() {},
  load() {
   return res.contentProtection.sessionsIDS;
  }
}
```

Reply:
The other pieces of information seems interesting to know for the user, except the `contentID`.